packages/meditrak-server/src/apiV2/export/exportSurveyResponses/exportResponsesToFile.js

```diff
@@ -0,0 +1,303 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import xlsx from 'xlsx';
+import moment from 'moment';
+import keyBy from 'lodash.keyby';
+import chunk from 'lodash.chunk';
+import {
+  addExportedDateAndOriginAtTheSheetBottom,
+  getExportDatesString,
+  getUniqueEntries,
+  truncateString,
+} from '@tupaia/utils';
+import { TYPES } from '@tupaia/database';
+import { ANSWER_TYPES, NON_DATA_ELEMENT_ANSWER_TYPES } from '../../../database/models/Answer';
+import { findAnswersInSurveyResponse, findQuestionsInSurvey } from '../../../dataAccessors';
+import { hasBESAdminAccess } from '../../../permissions';
+import { getExportPathForUser } from '../getExportPathForUser';
+import { zipMultipleFiles } from '../zipMultipleFiles';
+
+const FILE_PREFIX = 'survey_response_export';
+const MAX_RESPONSES_PER_FILE = 10000; // exporting too many responses in one file ends up out of memory
+export const EXPORT_DATE_FORMAT = 'D-M-YYYY h:mma';
+export const API_DATE_FORMAT = 'YYYY-MM-DD';
+const INFO_COLUMNS = {
+  id: 'Id',
+  type: 'Type',
+  code: 'Code',
+  text: 'Question',
+};
+
+function getEasyReadingInfoColumns(startDate, endDate) {
+  return { text: `Survey responses ${getExportDatesString(startDate, endDate)}` };
+}
+export const INFO_COLUMN_HEADERS = Object.values(INFO_COLUMNS); // For importer to understand format
+export const INFO_ROW_HEADERS = ['Entity Code', 'Entity Name', 'Date'];
+
+const getBlankWorkbook = () => ({ SheetNames: [], Sheets: {} });
+
+/**
+ * Exports excel documents containing the relevant survey responses, splitting into multiple files
+ * and zipping if too large.
+ *
+ * Note that this is a large function that is broken up into many internal helper functions, a bit
+ * like a class. The main body of the function is right at the bottom.
```

Comment:
I originally went for a class, but given that it is fundamentally just a loop with a short lifespan, and that it didn't really make sense to have some aspects passed as arguments to the constructor and others as parameters to the main entrypoint, I changed it to a function and found the mental model sat better. I'm happy for this to be vetoed though, if the layout is too unexpected.